### PR TITLE
[SYCL][DeviceSanitizer] Use -asan-constructor-kind=none to disable ctor/dtor

### PR DESCRIPTION
**Pull request by wenju-he**
The removed code fails to prevent ctor/dtor generation.

### CODE REVIEW

**Review by AlexeySachkov — APPROVED** [review]

File: clang/lib/Driver/SanitizerArgs.cpp [hunk 1]
```diff
@@ -1151,6 +1151,9 @@ void SanitizerArgs::addArgs(const ToolChain &TC, const llvm::opt::ArgList &Args,
       CmdArgs.push_back("-mllvm");
       CmdArgs.push_back("-asan-instrumentation-with-call-threshold=0");
 
+      CmdArgs.push_back("-mllvm");
+      CmdArgs.push_back("-asan-constructor-kind=none");
+
```
Comment (mdtoguchi) [inline clang/lib/Driver/SanitizerArgs.cpp:1156]: Please also update `clang/test/Driver/sycl-device-sanitizer.cpp`

File: sycl/test/check_device_code/device_asan_no_ctor.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// REQUIRES: linux
+// RUN: %clangxx -fsycl-device-only -fno-sycl-early-optimizations -Xarch_device -fsanitize=address -emit-llvm %s -S -o %t.ll
+// RUN: FileCheck %s --input-file %t.ll
+
+// CHECK-NOT: asan.module_ctor
```
Comment (aelovikov-intel) [inline sycl/test/check_device_code/device_asan_no_ctor.cpp:5]: Please add a comment explaining why we want that.

File: clang/lib/Driver/SanitizerArgs.cpp (hunk 1 above)
Comment (wenju-he) [reply clang/lib/Driver/SanitizerArgs.cpp:1157]: done

File: sycl/test/check_device_code/device_asan_no_ctor.cpp (hunk 2 above)
Comment (wenju-he) [reply sycl/test/check_device_code/device_asan_no_ctor.cpp:8]: done

**Review by AllanZyne — APPROVED** [review]

**Review by aelovikov-intel — APPROVED** [review]

**Comment by wenju-he** [comment]
@intel/llvm-gatekeepers please merge, thanks.

**Comment by ldrumm** [comment]
> @intel/llvm-gatekeepers please merge, thanks.

We still need @intel/dpcpp-clang-driver-reviewers

**Review by mdtoguchi — APPROVED** [review]
LGTM

**Pull request merged**